contracts/src/groth16.rs

```diff
@@ -12,35 +12,34 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use alloy_sol_types::SolValue;
-use anyhow::Result;
-use risc0_zkvm::{sha::Digestible, Groth16ReceiptVerifierParameters};
+use alloy::sol_types::SolValue;
+use risc0_zkvm::{Groth16Receipt, ReceiptClaim};
 
-/// ABI encoding of the seal.
-pub fn abi_encode(seal: Vec<u8>) -> Result<Vec<u8>> {
-    Ok(encode(seal)?.abi_encode())
+/// ABI encoding of a [Groth16Receipt][risc0_zkvm::Groth16Receipt] seal.
+pub fn abi_encode(receipt: &Groth16Receipt<ReceiptClaim>) -> Vec<u8> {
+    encode(receipt).abi_encode()
 }
 
-/// encoding of the seal with selector.
-pub fn encode(seal: Vec<u8>) -> Result<Vec<u8>> {
-    let verifier_parameters_digest = Groth16ReceiptVerifierParameters::default().digest();
-    let selector = &verifier_parameters_digest.as_bytes()[..4];
+/// Encoding of a [Groth16Receipt][risc0_zkvm::Groth16Receipt] by prefixing it with the verifier selector,
+/// taken from the first 4 bytes of the hash of the verifier parameters including the Groth16
+/// verification key and the control IDs that commit to the RISC Zero circuits.
+pub fn encode(receipt: &Groth16Receipt<ReceiptClaim>) -> Vec<u8> {
```

Comment:
How will we recommend going from https://docs.rs/bonsai-sdk/0.9.0/bonsai_sdk/responses/struct.SnarkReceipt.html, given we still recommend using bonsai-sdk over the zkvm for production use cases?